src/main/java/com/jcabi/log/ArrayFormatAction.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2012-2014, jcabi.com
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met: 1) Redistributions of source code must retain the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer. 2) Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.log;
+
+import com.jcabi.aspects.Immutable;
+import java.security.PrivilegedAction;
+import java.util.Arrays;
+import java.util.Formatter;
+
+/**
+ * {@link PrivilegedAction} for obtaining array contents.
+ * @author Aleksey Popov (alopen@yandex.ru)
+ * @version $Id$
+ */
+@Immutable
+final class ArrayFormatAction
+    implements PrivilegedAction<String>  {
+    /**
+     * Array to format.
+     */
+    private final transient Object[] array;
+
+    /**
+     * Constructor.
+     * @param arr Array to format
+     */
+    ArrayFormatAction(final Object[] arr) {
+        this.array = Arrays.copyOf(arr, arr.length);
+    }
+
+    @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public String run() {
+        final StringBuilder builder = new StringBuilder("[");
+        final Formatter formatter = new Formatter(builder);
+        for (int index = 0; index < this.array.length; index += 1) {
```

Comment:
why don't use same approach as in ObjectContentsFormatAction?
I mean, foreach and 
```builder.replace(...);``` in the end?
It should look better than indexes